lowranceusr.cc

```diff
@@ -355,34 +842,63 @@ lon_mm_to_deg(double x)
   return x / (DEGREESTORADIANS * SEMIMINOR);
 }
 
-static long
-lon_deg_to_mm(double x)
-{
-  return lround(x * SEMIMINOR * DEGREESTORADIANS);
-}
-
 static double
 lat_mm_to_deg(double x)
 {
   return (2.0 * atan(exp(x / SEMIMINOR)) - M_PI / 2.0) / DEGREESTORADIANS;
 }
 
+static long
+lon_deg_to_mm(double x)
+{
+  return (long)(x * SEMIMINOR * DEGREESTORADIANS);
+}
+
 static long
 lat_deg_to_mm(double x)
 {
   return (long)(SEMIMINOR * log(tan((x * DEGREESTORADIANS + M_PI / 2.0) / 2.0)));
 }
 
+
 static void
-lowranceusr_parse_waypt(Waypoint* wpt_tmp)
+lowranceusr_parse_waypt(Waypoint* wpt_tmp, int object_num_present)
 {
   char buff[MAXUSRSTRINGSIZE + 1];
 
+  /* Object num */
+  if (object_num_present) {
+    short object_num = gbfgetint16(file_in);
+    if (global_opts.debug_level > 2) {
+      if (global_opts.debug_level == 99) {
+        printf(MYNAME " parse_waypt: %5d", object_num);
+      } else {
+        printf(MYNAME " parse_waypt: object_num = %d\n", object_num);
+      }
+    }
+  }
+
   wpt_tmp->latitude = lat_mm_to_deg(gbfgetint32(file_in));
   wpt_tmp->longitude = lon_mm_to_deg(gbfgetint32(file_in));
-  wpt_tmp->altitude = FEET_TO_METERS(gbfgetint32(file_in));
-  if (METERS_TO_FEET(wpt_tmp->altitude) <= -10000) {
-    wpt_tmp->altitude = unknown_alt;
+  /* 
+   * From available data, looks like stream version = 0 used float to store for altitude in feet
+   * and version 1 used int32.  If that is found to not be the case may need to add flag for input.
+   * Only have a small sample size.
+   */
+  if (rstream_version == 0) {
+    float read_alt = gbfgetflt(file_in);
+    char buf[16];
```

Comment:
this hunk is weird. Using sprintf and strcmp to determine isnan isn't great. Is this basically
read_alt = getflt() (may rely on safe upcast of float todouble)
if isNaN(read_alt)) {
wpt_tmp->altitude = unknown_alt; (Steve, did we just not get optional to alt because of the zillion places it's used?)
} else {
  wpt_tmp->altitude = FTOMeters()
} else {
 wpt_tmp->altitude = ...
}